sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -515,10 +515,37 @@ object RemoveRedundantAliases extends Rule[LogicalPlan] {
  * Remove no-op operators from the query plan that do not make any modifications.
  */
 object RemoveNoopOperators extends Rule[LogicalPlan] {
+  def resolveOutputAttrName(
+      output: Seq[NamedExpression],
+      projectList: Seq[NamedExpression]): Seq[NamedExpression] = {
+    output.map {
+      case attr: Attribute =>
+        attr.withName(projectList.find(_.semanticEquals(attr)).getOrElse(attr).name)
```

Comment:
do we need to call `find`? `child.sameOutput(p)` should guarantee that the output columns match by position. We just need
```
def restoreOriginalOutputNames(
    projectList: Seq[NamedExpression],
    originalNames: Seq[String]): Seq[NamedExpression] = {
  projectList.zip(originalNames).map {
    case (attr: Attribute, name) => attr.withName(name)
    ...
  }
}
```

Reply:
In current approach don't need find

Reply:
Changed